# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/child/FileSystemRequestHandler.cpp
# Commit: 302fdb94bf7f
# Full Hash: 302fdb94bf7f5cd817b69fc4eaaaf6ff8b62bec8
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-02-06 21:36:23
# Regressor Bug: 1809064
# File Overlap Count: 1
# Description:
#   Bug 1809064 - Use managed endpoints for PFileSystemAccessHandle construction; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D166342
# ==============================================================================

diff -r 100b45bcf622 -r 302fdb94bf7f dom/fs/child/FileSystemRequestHandler.cpp
--- a/dom/fs/child/FileSystemRequestHandler.cpp	Mon Feb 06 15:01:20 2023 +0000
+++ b/dom/fs/child/FileSystemRequestHandler.cpp	Mon Feb 06 15:10:34 2023 +0000
@@ -26,6 +26,7 @@
 #include "mozilla/dom/IPCBlobUtils.h"
 #include "mozilla/dom/Promise.h"
 #include "mozilla/dom/quota/QuotaCommon.h"
+#include "mozilla/dom/quota/ResultExtensions.h"
 
 namespace mozilla::dom::fs {
 
@@ -123,14 +124,12 @@
     RefPtr<FileSystemManager>& aManager) {
   auto& properties = aResponse.get_FileSystemAccessHandleProperties();
 
-  auto* const actor =
-      static_cast<FileSystemAccessHandleChild*>(properties.accessHandleChild());
-
-  QM_TRY_UNWRAP(RefPtr<FileSystemSyncAccessHandle> result,
-                FileSystemSyncAccessHandle::Create(
-                    aGlobal, aManager, actor,
-                    std::move(properties.streamParams()), aMetadata),
-                nullptr);
+  QM_TRY_UNWRAP(
+      RefPtr<FileSystemSyncAccessHandle> result,
+      FileSystemSyncAccessHandle::Create(
+          aGlobal, aManager, std::move(properties.streamParams()),
+          std::move(properties.accessHandleChildEndpoint()), aMetadata),
+      nullptr);
 
   return result;
 }